Serilog.Sinks.Unity3D/Assets/Serilog.Sinks.Unity3D/UnitySinkExtensions.cs

```diff
@@ -52,12 +53,23 @@ public static class UnitySinkExtensions
             this LoggerSinkConfiguration sinkConfiguration,
             ITextFormatter formatter,
             LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
-            LoggingLevelSwitch levelSwitch = null)
+            LoggingLevelSwitch levelSwitch = null,
+            UnityEngine.ILogger logger = null)
         {
             if (sinkConfiguration == null) throw new ArgumentNullException(nameof(sinkConfiguration));
             if (formatter == null) throw new ArgumentNullException(nameof(formatter));
 
-            return sinkConfiguration.Sink(new Unity3DLogEventSink(formatter), restrictedToMinimumLevel, levelSwitch);
+            return sinkConfiguration.Sink(new Unity3DLogEventSink(formatter, logger), restrictedToMinimumLevel, levelSwitch);
+        }
+
+
+        public static ILogger WithUnityContext(this ILogger logger, UnityEngine.Object context)
+        {
+            int id = context.GetInstanceID();
+#if UNITY_EDITOR
+            Unity3DLogEventSink._objectstoLog.TryAdd(id, context);
```

Comment:
With Unity 2019.2.0 the Dictionary<T,R> type does not have the TryAdd method, and this does not compile in the editor
![image](https://user-images.githubusercontent.com/19240945/205995342-8b8bb80b-5ee9-4951-921d-d69d4d1feffc.png)